src/pages/PostAJob.js

```diff
@@ -1,24 +1,60 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import Layout from '../layouts/Layout'
+import { motion } from 'framer-motion'
+import { loadStripe } from '@stripe/stripe-js'
+
+//components
 import PostAJobForm from '../components/form/PostAJobForm'
 import StatusBar from '../components/form/StatusBar'
 import JobTemplate from '../components/JobTemplate'
 import JobPostingConfirmation from '../components/JobPostingConfirmation'
-import { motion } from 'framer-motion'
-import Layout from '../layouts/Layout'
+import TierSelect from '../components/form/TierSelect'
 
+//firebase
 import { db, storage } from '../firebase/firebase'
 import firebase from 'firebase'
 
-const PostAJob = () => {
-  const [status, setStatus] = useState(1)
+const stripePromise = loadStripe(
+  process.env.REACT_APP_STRIPE_API_KEY
+)
+
+const PostAJob = ({ location }) => {
+
+  const tierQueryParam = findParam('t') ? findParam('t').split('=')[1] : process.env.REACT_APP_ADVANCED_PLAN
+
+  let history = useHistory()
+
+  const [status, setStatus] = useState()
```

Comment:
There is a lot of `useState` hooks here. Consider in this PR or a subsequent PR to use the [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer) hook instead.

Reply:
I've always claimed to have never run into a use case for useReducer. That may partly be true because I don't fully understand what it does. I'll read into useReducer and refactoring this.

Reply:
Like I said, consider, so not critical. At some point you might realize that too many useStates becomes unwieldy but if you don’t feel like that, carry on. 😎

Reply:
Hey @nickytonline I'm gonna leave this for now, but I'd LOVE to chat with you about this and the other places you feel could be improved (with useReducer or really any way).